src/cascadia/TerminalControl/TermControl.h

```diff
@@ -35,6 +35,9 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         winrt::fire_and_forget UpdateControlSettings(Control::IControlSettings settings, Control::IControlAppearance unfocusedAppearance);
         IControlSettings Settings() const;
 
+        TerminalConnection::ConnectionInformation ConnectionInfo();

```

Comment:
I don't totally understand why TerminalPage can't just _make a totally new connection out of nothing_ - I find it so weird that TermControl now has a Connection (which we gave it) and a ConnectionInfo (_which we used to make the Connection_, and then also gave it). How would a public user of TermControl's public API even know what to do with all of these members?

Reply:
I don't love the pattern of storing effectively random app data on random other classes, you know? We did the same for Profile (and pushed it all the way down to the Core)...